framework/src/utils/RankTwoTensor.C

```diff
@@ -580,6 +594,56 @@ RankTwoTensor::mixedProductJkIl(const RankTwoTensor & b) const
   return result;
 }
 
+RankFourTensor
+RankTwoTensor::positveProjectionEigenDecomposition() const
+{
+  // The calculate of projection tensor follows
+  // C. Miehe and M. Lambrecht, Commun. Numer. Meth. Engng 2001; 17:337~353
+
+  // Compute eigenvectors and eigenvalues of mechanical strain
+  RankTwoTensor eigvec;
+  std::vector<Real> eigval(N);
+  (*this).symmetricEigenvaluesEigenvectors(eigval, eigvec);
```

Comment:
the `(*this).` (which should always be written as `this->` for clarity) is not needed here. It'd be neat fo we could change the vector above to an array as well, but that might require templating or overloading the `symmetricEigenvaluesEigenvectors` member. Let's do those optimizations in a different PR.

Reply:
Sounds great. I will address them in next PR.